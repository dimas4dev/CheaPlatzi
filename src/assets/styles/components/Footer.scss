@import "../vars.scss";
.footer {
    display: flex;
    justify-content: space-evenly;
    align-content: center;
    min-height: 90px;
    background: rgb(2, 0, 36);
    background: linear-gradient( 90deg, rgba(2, 0, 36, 1) 0%, rgba(9, 9, 121, 1) 35%, rgba(0, 212, 255, 1) 100%);
    padding: 20px 60px;
    &__contact {
        p {
            color: $color-white;
            font-size: 20px;
            width: 50%;
        }
    }
    &__social {
        display: flex;
        align-items: center;
        figure {
            margin-left: 28px;
            cursor: pointer;
        }
    }
}

@media (max-width: 900px) {
    .footer {
        padding: 20px 30px;
    }
    .footer__social figure {
        margin-left: 18px;
    }
}

@media (max-width: 768px) {
    .footer {
        padding: 1em 2em;
        flex-direction: column;
        align-items: center;
        text-align: center;
    }
    .footer__social {
        margin: 2em 0 1em 0;
    }
    .footer__contact {
        p {
            text-align: center;
            width: auto;
        }
    }
}

@media (max-width: 425px) {
    .footer {
        padding: 1em 1em;
    }
    .footer__contact {
        p {
            text-align: center;
            width: auto;
        }
    }
}