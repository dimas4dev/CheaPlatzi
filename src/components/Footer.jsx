import React from "react";

import Contact from "./Contact";

import instagram from "../assets/images/instagram.svg";
import facebook from "../assets/images/facebook.svg";
import "../assets/styles/components/Footer.scss";

const Footer = () => {
  return (
    <footer className="footer">
      <div className="footer__contact">
        <p>Cont√°ctanos y resolveremos todas tus inquietudes</p>
        <Contact
          guide="Tu email"
          action="Enviar"
          styles="button--search button--yellow"
        />
      </div>
      <div className="footer__social">
        <figure>
          <img src={instagram} alt="link to Instagram" />
        </figure>
        <figure>
          <img src={facebook} alt="link to facebook" />
        </figure>
      </div>
    </footer>
  );
};

export default Footer;
