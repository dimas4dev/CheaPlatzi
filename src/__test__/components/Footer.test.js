import React from 'react';
import {mount} from 'enzyme';
import {create} from 'react-test-renderer';
import Footer from '../../components/Footer';

describe('<Footer />', () => {
  const footer = mount(<Footer />);

  test('Render del componente Footer', () => {
    expect(footer.length).toEqual(1);
  });
  test('Render del titulo', () => {
    expect(footer.find('.footer__contact--label').text()).toEqual(
      'ContÃ¡ctanos y resolveremos todas tus inquietudes'
    );
  });
});

describe('Footer Snapshot', () => {
  test('Comprobar la UI del componente Footer', () => {
    const footer = create(<Footer />);
    expect(footer.toJSON()).toMatchSnapshot();
  });
});
