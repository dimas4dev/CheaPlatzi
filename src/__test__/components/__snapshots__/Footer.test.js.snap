// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Comprobar la UI del componente Footer 1`] = `
<footer
  className="footer"
>
  <div
    className="footer__contact"
  >
    <label
      className="footer__contact--label"
      htmlFor="contact"
    >
      Cont√°ctanos y resolveremos todas tus inquietudes
    </label>
    <form
      onSubmit={[Function]}
    >
      <input
        className="contact__input"
        id="contact"
        placeholder="Tu email"
        type="text"
      />
      <button
        className="button--search button--yellow"
        type="search"
      >
        Enviar
      </button>
    </form>
  </div>
  <div
    className="footer__social"
  >
    <figure>
      <img
        alt="link to Instagram"
        src={Object {}}
      />
    </figure>
    <figure>
      <img
        alt="link to facebook"
        src={Object {}}
      />
    </figure>
  </div>
</footer>
`;
